cargo-dist/src/tasks.rs

```diff
@@ -49,50 +92,221 @@ pub const CPU_ARM64: &str = "arm64";
+    /// us to build `x86_64-apple-darwin` and `aarch64-apple-darwin` (arm64) and then combine
+    /// them into a "universal" binary that can run on either arch (using apple's `lipo` tool).
+    ///
+    /// FIXME: Allow higher level requests like "[macos, windows, linux] x [x86_64, aarch64]"?
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub targets: Option<Vec<String>>,
+
+    /// Include the following static files in bundles like executable-zips.
+    ///
+    /// Paths are relative to the Cargo.toml this is defined in.
+    ///
+    /// Files like `README*`, `(UN)LICENSE*`, `RELEASES*`, and `CHANGELOG*` are already
+    /// automatically detected and included (use [`DistMetadata::auto_includes`][] to prevent this).
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub include: Vec<Utf8PathBuf>,
+
+    /// Whether to auto-include files like `README*`, `(UN)LICENSE*`, `RELEASES*`, and `CHANGELOG*`
+    ///
+    /// Defaults to true.
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "auto-includes")]
+    pub auto_includes: Option<bool>,
+}
+
+impl DistMetadata {
+    /// Apply the base path to any relative paths contained in this DistMetadata
+    fn make_relative_to(&mut self, base_path: &Utf8Path) {
+        for include in &mut self.include {
+            *include = base_path.join(&*include);
+        }
+    }
+
+    fn merge_workspace_config(&mut self, workspace_config: &Self) {
+        // These fields omitted because they should only be set globally at the workspace root
+        // and we should just error if you set them multiple times or locally!
+        //
+        // * cargo_dist_version
+        // * rust_toolchain_version
+        // * ci
+
+        if self.installers.is_none() {
+            self.installers = workspace_config.installers.clone();
+        }
+        if self.targets.is_none() {
+            self.targets = workspace_config.targets.clone();
+        }
+        if self.dist.is_none() {
+            self.dist = workspace_config.dist;
+        }
+        if self.auto_includes.is_none() {
+            self.auto_includes = workspace_config.auto_includes;
+        }
+        self.include
+            .extend(workspace_config.include.iter().cloned());
+    }
+}
+
+/// Parts of a [profile.*] entry in a Cargo.toml we care about
+#[derive(Debug, Clone)]
+pub struct CargoProfile {
+    /// Whether debuginfo is enabled.
+    ///
+    /// can be 0, 1, 2, true (=2), false (=0).
+    pub debug: Option<i64>,
+    /// Whether split-debuginfo is enabled.
+    ///
+    /// Can be "off", "packed", or "unpacked".
+    ///
+    /// If "packed" then we expect a pdb/dsym/dwp artifact.
+    pub split_debuginfo: Option<String>,
+}
 
 /// Global config for commands
 #[derive(Debug)]
 pub struct Config {
     /// Whether we'll actually run builds (if false we'll still generate installers)
```

Comment:
Just a quick note that those docs are now outdated.

Reply:
good catch